### PR TITLE
Feat/BNGP-3186-LegalAgreement2NewPages

### PR DESCRIPTION
**Pull request by KiranVarma02**
https://eaflood.atlassian.net/jira/software/c/projects/BNGP/boards/1409?selectedIssue=BNGP-3186

### CODE REVIEW

File: bng.code-workspace [hunk 1]
```diff
@@ -14,7 +14,11 @@
 		}
 	],
 	"settings": {
-		"debug.internalConsoleOptions": "neverOpen"
+		"debug.internalConsoleOptions": "neverOpen",
+		"editor.codeActionsOnSave": {
+			"source.fixAll": true
+		},
+		"eslint.validate": ["javascript", "html"]
```
Comment (KiranVarma02) [inline bng.code-workspace:21]: Note: Auto fix standard lint rules on save

File: packages/webapp/src/routes/__tests__/land/legal-party-list.spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import { submitGetRequest } from '../helpers/server.js'
+import constants from '../../../utils/constants.js'
+const url = constants.routes.LEGAL_PARTY_LIST
+
+describe('url', () => {
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/legal-party-list.spec.js:5]: Probably want to describe `url` and not the string 'url'

File: packages/webapp/src/routes/__tests__/land/legal-party-list.spec.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    ])
+
+    legalPartyList = require('../../land/legal-party-list.js')
+  })
+
+  describe('GET', () => {
+    it(`should render the ${url.substring(1)} view`, async () => {
+      const response = await submitGetRequest({ url })
+      expect(response.statusCode).toBe(200)
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/legal-party-list.spec.js:43]: By default submitGetRequest will already assert a 200 response, unless a different http code is passed through

File: packages/webapp/src/routes/__tests__/land/legal-party-remove.spec.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import { submitGetRequest } from '../helpers/server.js'
+import constants from '../../../utils/constants.js'
+const url = constants.routes.LEGAL_PARTY_REMOVE
+
+describe('url', () => {
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/legal-party-remove.spec.js:5]: `url` again

**Review by teddmason — CHANGES_REQUESTED** [review]
As per slack message the following issues need testing/sorting to ensure compatibility with new legal agreement parties array format:
- /land/check-legal-agreement-details
- /land/check-and-submit
- application.js and applicaiton-schema.js

File: packages/webapp/src/routes/__tests__/land/legal-party-list.spec.js (hunk 2 above)
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/legal-party-list.spec.js:5]: duh, yes ofcourse :)

File: packages/webapp/src/views/land/legal-party-list.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    {% if err %}
+        {% include "errorbanner.html" %}
+    {% endif %}
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            <h1 class="govuk-heading-l" tabindex="-1">
+              {{ pageHeading }}
+            </h1>
+
+            {% for org in legalAgreementParties %}
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/land/legal-party-list.html:22]: This creates a new govSummaryList for every legal agreement party, which messes up the formatting. You should have one govSummaryList with one row per legal agreement party. You can achieve this by passing the row array data already processed into the template from the route handler (i.e. put this logic in the route handler not the template).

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Couple of minor things to address

**Review by paulsimonandrews — DISMISSED** [review]
LGTM

**Review by paulsimonandrews — APPROVED** [review]
LGTM

**Comment by shaundefra** [comment]
Tedd is on leave and as Paul Andrews has reviewed this am happy to override the branch protection for this merge
tested and merged pending review from PO
https://defra-digital.slack.com/archives/C0473GYG80L/p1691567863740149

**Pull request merged**